homeassistant/components/iotawatt/sensor.py

```diff
@@ -164,11 +164,7 @@ def __init__(
         super().__init__(coordinator=coordinator)
 
         self._key = key
-        data = self._sensor_data
-        if data.getType() == "Input":
-            self._attr_unique_id = (
-                f"{data.hub_mac_address}-input-{data.getChannel()}-{data.getUnit()}"
-            )
+        self._attr_unique_id = self._sensor_data.getSensorID()
```

Comment:
This returns a sensor names based on the name of the sensor. 

A name is not a valid unique identifier for Home Assistant.

See: <https://developers.home-assistant.io/docs/entity_registry_index?unique-id-requirements>